### PR TITLE
Fix to use updated siteUrlInput

### LINKED ISSUE #14474: Signup: cannot add an existing WordPress site
Steps to reproduce:

1. Log in to https://wordpress.com/
1. Go to My Sites > Switch Site > Add New Site.
1. Try to type into the siteUrl field to add an existing WordPress site with Jetpack.

<strong>Result:</strong> cannot enter text into the siteUrl field and there are no errors in the console.

Note: you may need to force the AB test to load, see p58i-5KQ-p2.

screen shot 2017-05-25 at thu may 25 8 23 14 am
<sup>Seen at https://wordpress.com/jetpack/new?ref=calypso-selector using Chrome 58.0.3029.110 on Mac OS X 10.12.4 logged in as a superadmin proxied.</sup>

screen shot 2017-05-25 at thu may 25 8 29 40 am
<sup>Seen at https://wordpress.com/jetpack/new?ref=calypso-selector using Firefox 53.0.3 on Mac OS X 10.12.4 logged in as design5279 unproxied.</sup>

Tested both staging and production.

/hat tip @molszewska for the original report at p1495692412741621-slack-triage

### PR DESCRIPTION
**Pull request by sirreal**
Fixes #14474

To test
* `localStorage.setItem( 'ABTests', '{"newSiteWithJetpack_20170419":"showNewJetpackSite"}' );`
* Go to `/jetpack/new`
* Ensure url inputs and buttons work correctly. Check mobile view as well.
* Ensure bug in #14474 is fixed

### CODE REVIEW

File: client/jetpack-connect/jetpack-new-site/index.jsx [hunk 1]
```diff
@@ -92,9 +95,13 @@ class JetpackNewSite extends Component {
 							<div className="jetpack-new-site__card-description">
 								{ this.props.translate( 'We’ll be using the Jetpack plugin to connect your site to WordPress.com.' ) }
 							</div>
-							<SiteURLInput ref="siteUrlInputRef"
-								onTosClick={ this.handleOnClickTos }
-								onClick={ this.onURLEnter } />
+							<SiteURLInput
+								ref="siteUrlInputRef"
```
Comment (seear) [inline client/jetpack-connect/jetpack-new-site/index.jsx]: ref still needed?

Comment (sirreal) [inline client/jetpack-connect/jetpack-new-site/index.jsx]: Shouldn't be, but doesn't hurt. I'd like to deploy and handle extra clean up later 😅

Comment (sirreal) [inline client/jetpack-connect/jetpack-new-site/index.jsx]: Scratch that, quick enough to remove.

**Comment by seear** [comment]
Kinda messy how entering a URL here takes you straight through to another instantiation of the same component, but we can clear that up another time.

**Pull request merged**